Index: trunk/lib/cpp/src/thrift/concurrency/BoostMonitor.cpp
===================================================================
--- trunk/lib/cpp/src/thrift/concurrency/BoostMonitor.cpp	(revision 1381333)
+++ trunk/lib/cpp/src/thrift/concurrency/BoostMonitor.cpp	(working copy)
@@ -114,8 +114,8 @@
     struct timespec currenttime;
     Util::toTimespec(currenttime, Util::currentTime());
 
-	long tv_sec = abstime->tv_sec - currenttime.tv_sec;
-	long tv_nsec = abstime->tv_nsec - currenttime.tv_nsec;
+	long tv_sec = THRIFT_SAFE_CAST(long, abstime->tv_sec - currenttime.tv_sec);
+	long tv_nsec = THRIFT_SAFE_CAST(long, abstime->tv_nsec - currenttime.tv_nsec);
 	if(tv_sec < 0)
 		tv_sec = 0;
 	if(tv_nsec < 0)
