Index: trunk/lib/cpp/src/thrift/server/TThreadPoolServer.cpp
===================================================================
--- trunk/lib/cpp/src/thrift/server/TThreadPoolServer.cpp	(revision 1381333)
+++ trunk/lib/cpp/src/thrift/server/TThreadPoolServer.cpp	(working copy)
@@ -72,7 +72,7 @@
           break;
         }
       }
-    } catch (const TTransportException& ttx) {
+    } catch (const TTransportException& ) {
       // This is reasonably expected, client didn't send a full request so just
       // ignore him
       // string errStr = string("TThreadPoolServer client died: ") + ttx.what();
