Index: trunk/lib/cpp/src/thrift/transport/TSocket.h
===================================================================
--- trunk/lib/cpp/src/thrift/transport/TSocket.h	(revision 1381333)
+++ trunk/lib/cpp/src/thrift/transport/TSocket.h	(working copy)
@@ -22,6 +22,13 @@
 
 #include <string>
 
+#if HAVE_SYS_SOCKET_H
+# include <sys/socket.h>
+#endif
+#if HAVE_NETINET_IN_H
+# include <netinet/in.h>
+#endif
+
 #include "TTransport.h"
 #include "TVirtualTransport.h"
 #include "TServerSocket.h"
@@ -259,8 +266,11 @@
   std::string path_;
 
   /** Underlying UNIX socket handle */
+#ifdef _WIN64not
+  SOCKET socket_;
+#else
   int socket_;
-
+#endif
   /** Connect timeout in ms */
   int connTimeout_;
 
